libbeat/processors/decode_xml_fields/decode_xml_fields.go

```diff
@@ -0,0 +1,164 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package decode_xml_fields
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/pkg/errors"
+
+	"github.com/elastic/beats/v7/libbeat/beat"
+	"github.com/elastic/beats/v7/libbeat/beat/events"
+	"github.com/elastic/beats/v7/libbeat/common"
+	"github.com/elastic/beats/v7/libbeat/common/jsontransform"
+	"github.com/elastic/beats/v7/libbeat/logp"
+	"github.com/elastic/beats/v7/libbeat/processors"
+	"github.com/elastic/beats/v7/libbeat/processors/checks"
+	jsprocessor "github.com/elastic/beats/v7/libbeat/processors/script/javascript/module/processor"
+)
+
+type decodeXMLFieldsConfig struct {
+	Fields        []string `config:"fields"`
+	OverwriteKeys bool     `config:"overwrite_keys"`
+	AddErrorKey   bool     `config:"add_error_key"`
+	Target        *string  `config:"target"`
```

Comment:
why pointer? What if `Target` is the empty string?

Reply:
This is mostly because I was following `decode_json_fields`, which implements it similarly, and also supports empty string as target field.

I also didn't really see a usage for it, but kept it for the sake of being as similar as possible to the other processor.

Reply:
@urso After getting some background info from Andrew, I see now why it was a pointer in the older decode_json_fields processor, I will change this to not be a pointer anymore, it was mostly to detect the empty string if nothing was set, but better to set it as part of the default config.

Reply:
I ended up having to revert this, as it was not really feasible to differentiate between nil and an empty string without using it as a pointer.